frontend/src/concepts/modelCatalog/content/ModelCatalogLabels.tsx

```diff
@@ -3,14 +3,27 @@ import { Label, LabelGroup } from '@patternfly/react-core';
 import { getILabLabels, removeILabLabels } from '~/pages/modelCatalog/utils';
 import { ReservedILabLabel } from '~/pages/modelCatalog/const';
 
-export const ModelCatalogLabels: React.FC<{
+const TRUNCATED_MAX_LABEL_NUM = 2;
+
+type ModelCatalogLabelsProps = {
   labels?: string[];
   tasks?: string[];
   showNonILabLabels?: boolean;
-}> = ({ labels = [], tasks = [], showNonILabLabels = false }) => (
-  <LabelGroup data-testid="model-catalog-label-group">
-    {getILabLabels(labels).map((l) => {
-      switch (l) {
+  truncate?: boolean;
```

Comment:
Instead of passing the `truncate` and set it as `TRUNCATED_MAX_LABEL_NUM` in the component, why not just pass a prop named `numLabels`? Then you can pass `numLabels={TRUNCATED_MAX_LABEL_NUM}` when using it and don't need to pass anything if you don't want the labels to be truncated. Also, it provides more flexibility because we can define the truncate number now for future use